crypt.go

```diff
@@ -928,9 +935,36 @@ func RenameLUKS2ContainerKey(devicePath, oldName, newName string) error {
 		return errors.New("cannot rename key with unexpected token type")
 	}
 
-	if err := luks2ImportToken(devicePath, newToken, &luks2.ImportTokenOptions{Id: id, Replace: true}); err != nil {
-		return xerrors.Errorf("cannot import new token: %w", err)
+	if replace {
+		if err := luks2ImportToken(devicePath, newToken, &luks2.ImportTokenOptions{Id: id, Replace: true}); err != nil {
+			return xerrors.Errorf("cannot import new token: %w", err)
+		}
+	} else {
+		if nonProliferation == nil {
+			return errors.New("internal error: using a functionality that should proliferate has been attempted")
+		}
+		if err := luks2ImportToken(devicePath, newToken, nil); err != nil {
+			return xerrors.Errorf("cannot import new token: %w", err)
+		}
+		if err := luks2RemoveToken(devicePath, id); err != nil {
+			return xerrors.Errorf("cannot remove existing token %d: %w", id, err)
+		}
 	}
 
 	return nil
 }
+
+// RenameLUKS2Container key renames the keyslot with the specified oldName on
+// the LUKS2 container at the specified path.
+func RenameLUKS2ContainerKey(devicePath, oldName, newName string) error {
+	const replace = true
+	return renameLUKS2ContainerKey(nil, devicePath, oldName, newName, replace)
+}
+
+// CopyAndRemoveLUKS2ContainerKey key renames the keyslot with the
+// specified oldName on the LUKS2 container at the specified path. It
+// does not use new feature from cryptsetup. However it is not atomic.
```

Comment:
To add here:

This must be used with care and intentionally. AllowNontAtomicOperation result must be provided to allow for its use.